packages/swap/src/quoting/Quoter.ts

```diff
@@ -58,28 +60,44 @@ export type RpcLimitOrder = {
   };
 };
 
-const formatLimitOrders = (
-  quotes: { orders: MarketMakerQuote['legs'][number] }[],
-  leg?: MarketMakerQuoteRequest<LegJson>['legs'][number],
-): RpcLimitOrder[] => {
-  if (!leg) return [];
-
-  return quotes.flatMap(({ orders }) =>
-    orders.map(([tick, amount]) => ({
-      LimitOrder: {
-        side: leg.side === 'BUY' ? 'sell' : 'buy',
-        base_asset: leg.base_asset,
-        quote_asset: leg.quote_asset,
-        tick,
-        sell_amount: hexEncodeNumber(amount),
-      },
-    })),
-  );
+const isBalanceWithinTolerance = (balance: bigint, amount: bigint) => {
+  const tolerance = BigInt(env.QUOTER_BALANCE_TOLERANCE_PERCENT);
+  let paddedBalance = balance;
+  if (tolerance > 0n) paddedBalance += (balance * tolerance) / 100n;
+  return paddedBalance >= amount;
 };
 
+function* formatLimitOrders(
+  quotes: { orders: MarketMakerQuote['legs'][number]; balances?: InternalAssetMap<bigint> }[],
+  leg?: MarketMakerQuoteRequest<LegJson>['legs'][number],
+): IteratorObject<RpcLimitOrder> {
+  if (!leg) return;
+
+  const sellAsset = getInternalAsset(leg.side === 'BUY' ? leg.base_asset : leg.quote_asset);
```

Comment:
we are sure that `leg.side` is from the perspective or the swapper, right? 😅

Reply:
yeah, you can see in the test that the USDC balance is checked when the source asset is BTC